hnn_core/network.py

```diff
@@ -389,8 +392,18 @@ def __init__(self, params, add_drives_from_params=False,
         self.pos_dict = dict()
         self.cell_types = dict()
 
-        self._N_pyr_x = self._params['N_pyr_x']
-        self._N_pyr_y = self._params['N_pyr_y']
+        # set the mesh shape
+        _validate_type(mesh_shape, tuple, 'mesh_shape')
+        _validate_type(mesh_shape[0], int, 'mesh_shape[0]')
+        _validate_type(mesh_shape[1], int, 'mesh_shape[1]')
```

Comment:
Is there a good reason to use this vs `isinstance`?

Reply:
Oh I see it's an internal hnn validation function. I thought it was a hidden function from an external mne package. Perhaps a future PR could move this function to a more sensible module if it is to be used widely?